packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormSelectFieldInput.tsx

```diff
@@ -0,0 +1,266 @@
+      throw new Error('Can only be called when editing a static value');
+    }
+
+    setDraftValue({
+      ...draftValue,
+      editingMode: 'view',
+    });
+
+    goBackToPreviousHotkeyScope();
+  };
+
+  const [selectWrapperRef, setSelectWrapperRef] =
+    useState<HTMLDivElement | null>(null);
+
+  const [filteredOptions, setFilteredOptions] = useState<SelectOption[]>([]);
+
+  const { resetSelectedItem } = useSelectableList(
+    SINGLE_RECORD_SELECT_BASE_LIST,
+  );
+
+  const clearField = () => {
+    setDraftValue({
+      type: 'static',
+      editingMode: 'view',
+      value: '',
+    });
+  };
+
+  const selectedOption = field.metadata.options.find(
+    (option) => option.value === draftValue.value,
+  );
+
+  const handleClearField = () => {
+    clearField();
+    onCancel();
+  };
+
+  const handleSubmit = (option: SelectOption) => {
+    onSubmit(option?.value);
+
+    resetSelectedItem();
+  };
+
+  const handleUnlinkVariable = () => {
+    setDraftValue({
+      type: 'static',
+      value: '',
+      editingMode: 'view',
+    });
+
+    onPersist(null);
+  };
+
+  const handleVariableTagInsert = (variableName: string) => {
+    setDraftValue({
+      type: 'variable',
+      value: variableName,
+    });
+
+    onPersist(variableName);
+  };
+
+  const handleDisplayModeClick = () => {
+    if (draftValue.type !== 'static') {
+      throw new Error(
+        'This function can only be called when editing a static value.',
+      );
+    }
+
+    setDraftValue({
+      ...draftValue,
+      editingMode: 'edit',
+    });
+
+    setHotkeyScopeAndMemorizePreviousScope(hotkeyScope);
+  };
+
+  const handleSelectEnter = (itemId: string) => {
+    const option = filteredOptions.find((option) => option.value === itemId);
+    if (isDefined(option)) {
+      onSubmit(option.value);
+      resetSelectedItem();
+    }
+  };
+
+  useScopedHotkeys(
+    Key.Escape,
+    () => {
+      onCancel();
+      resetSelectedItem();
+    },
+    hotkeyScope,
+    [onCancel, resetSelectedItem],
+  );
+
+  const optionIds = [
+    `No ${field.label}`,
+    ...filteredOptions.map((option) => option.value),
+  ];
+
+  return (
+    <StyledFormFieldInputContainer>
+      {label ? <InputLabel>{label}</InputLabel> : null}
+
+      <StyledFormFieldInputRowContainer>
+        <StyledFormFieldInputInputContainer
+          ref={setSelectWrapperRef}
+          hasRightElement={isDefined(VariablePicker)}
+        >
+          {draftValue.type === 'static' ? (
+            <>
+              <StyledDisplayModeContainer
+                data-open={draftValue.editingMode === 'edit'}
+                onClick={handleDisplayModeClick}
+              >
+                <VisibilityHidden>Edit</VisibilityHidden>
+
+                {isDefined(selectedOption) ? (
+                  <Tag
+                    preventShrink
+                    color={selectedOption.color}
+                    text={selectedOption.label}
+                  />
```

Comment:
can you use `<SelectFieldDisplay` component here?

Reply:
we might create a new components to decouple the component design definition from the fieldContext. Thus we will be able to factorize

Reply:
I created a component to unify what we display for the display mode of a select input.

Reply:
nice, is it possible to do the same with the input mode?

Reply:
Would you like to extract this part?

```tsx
<SelectableList
  selectableListId={SINGLE_RECORD_SELECT_BASE_LIST}
  selectableItemIdArray={optionIds}
  hotkeyScope={hotkeyScope}
  onEnter={(itemId) => {
    const option = filteredOptions.find(
      (option) => option.value === itemId,
    );
    if (isDefined(option)) {
      onSubmit?.(() => persistField(option.value));
      resetSelectedItem();
    }
  }}
>
  <SelectInput
    parentRef={selectWrapperRef}
    onOptionSelected={handleSubmit}
    options={fieldDefinition.metadata.options}
    onCancel={onCancel}
    defaultOption={selectedOption}
    onFilterChange={setFilteredOptions}
    onClear={
      fieldDefinition.metadata.isNullable ? handleClearField : undefined
    }
    clearLabel={fieldDefinition.label}
    hotkeyScope={hotkeyScope}
  />
</SelectableList>
```

I'm not sure it would be great, as I would end up with a component taking +10 unrelated props.

Reply:
For me, the `SelectableList` and `SelectInput` components are well enough abstracted. They own the design decisions. These two components are meta components with business logic. I don't think we need to go deeper into the refactoring.

Reply:
Like this, the way to display or modify a SELECT value is uniquely defined throughout the application